### PR TITLE
Show track codes, timeline improvements

### PR DESCRIPTION
**Pull request by pferreir**
Missing:
 * [x] PDF files

### CODE REVIEW

File: indico/modules/events/abstracts/templates/management/_abstract_list.html [hunk 1]
```diff
@@ -101,35 +129,24 @@
...
+                                        {{ _render_multi_track_column(abstract.reviewed_for_tracks) }}
                                     {% elif item.id == 'accepted_track' %}
-                                        {% set track = abstract.accepted_track.title if abstract.accepted_track else '' %}
-                                        <td class="i-table" data-text="{{ track }}"
-                                            data-searchable="{{ track | lower }}">
+                                        {% set track = abstract.accepted_track %}
+                                        {% if track %}
+                                            {% set track_text = track.short_title %}
+                                            {% set track_full = _get_track_full_title(track) %}
+                                            {% set track_searchable = _get_track_full_title(track, searchable=true) | lower%}
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/management/_abstract_list.html]: missing space after `lower`

File: indico/modules/events/abstracts/templates/abstract/_common.html [hunk 2]
```diff
@@ -19,14 +19,33 @@
...
 {% endmacro %}
 
-{% macro render_tracks(tracks) %}
-    {% for track in tracks %}
-        <span class="abstract-track">{{ track.title }}</span>
+{% macro render_track(track, full_title=false, classes='abstract-track') -%}
+    {% set title = track.full_title if full_title else track.short_title %}
+    {% if track.can_review_abstracts(session.user) or track.can_convene(session.user) %}
+        <a class="{{ classes }}"
+           href="{{ url_for('.display_reviewable_track_abstracts', track) }}"
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/_common.html]: href before class

File: indico/modules/events/abstracts/templates/abstract/timeline.html [hunk 3]
```diff
@@ -74,33 +76,79 @@
...
+                                    Proposed to {{ action }}Accept{{ endaction }}
+                                {%- endtrans %}
+                            {% endif %}
+                        {% elif review.proposed_action.name == 'reject' %}
                             {% trans action=_render_review_state(review) -%}
-                                Proposes to {{ action }}
+                                Proposed to {{ action }}Reject{{ endaction }}
+                            {%- endtrans %}
+                        {% elif review.proposed_action.name == 'change_tracks' %}
+                            {% trans type=contribution_type|safe -%}
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: `type` isn't used

File: indico/modules/events/abstracts/templates/abstract/timeline.html [hunk 4]
```diff
@@ -74,33 +76,79 @@
...
+                        {% if review.proposed_action.name == 'accept' %}
+                            {% if review.proposed_contribution_type %}
+                                {% set contribution_type -%}
+                                    <strong>{{ review.proposed_contribution_type.name|escape }}</strong>
+                                {%- endset %}
+                                {% trans type=contribution_type|safe -%}
+                                    Proposed to {{ action }}Accept{{ endaction }} as {{ type }}
+                                {%- endtrans %}
+                            {% else %}
+                                {% trans action=_render_review_state(review) -%}
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: already set, no need to repeat this here

File: indico/modules/events/abstracts/templates/abstract/timeline.html [hunk 5]
```diff
@@ -74,33 +76,79 @@
...
+                            {% set other_abstract = review.proposed_related_abstract %}
+                            {% set abstract_link -%}
+                                <a href="{{ url_for('.display_abstract', other_abstract) }}">
+                                    {{ other_abstract.title|escape }}
+                                </a>
+                            {%- endset %}
+                            {% set title = abstract_link|safe %}
+                            {% if review.proposed_action.name == 'merge' %}
+                                {% trans -%}
+                                    Proposed to {{ action }}Merge{{ endaction }} into {{ title }}
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: lowercase merge?

File: indico/modules/events/abstracts/templates/abstract/timeline.html [hunk 6]
```diff
@@ -112,81 +160,53 @@
...
-                        {% trans type=contribution_type|safe -%}
-                            Proposed to accept as {{ type }}
-                        {%- endtrans %}
-                    {% else %}
-                        {% trans -%}
-                            Proposed to accept
+                <div class="i-box-content js-form-container">
+                    {% if review.proposed_action.name == 'change_tracks' %}
+                        {% set tracks = render_tracks(review.proposed_tracks) %}
+                        {% trans count=review.proposed_tracks|length, tracks=tracks -%}
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: `tracks=tracks` is not needed, plain variables work fine without specifying them here

File: indico/modules/events/abstracts/templates/abstract/timeline.html [hunk 7]
```diff
@@ -112,81 +160,53 @@
...
-                            </div>
-                            <div>
-                                <span class="value">
-                                    {{ _render_rating(review.abstract.event_new.cfa, rating.value) }}
-                                    <span class="number">{{ rating.value }}</span>
-                                </span>
+                    {% if review.ratings %}
+                        <div class="ratings-switch">
+                            <a class="i-button"
+                               title="{% trans %}Show detailed ratings{% endtrans %}">...</a>
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: fits nicely in one line

File: indico/modules/events/abstracts/templates/abstract/timeline.html [hunk 8]
```diff
@@ -112,81 +160,53 @@
...
-                {% if review.comment %}
-                    <div class="titled-rule">
-                        {% trans %}Comment{% endtrans %}
-                    </div>
-                    <div class="markdown-text">
-                        {{ review.comment }}
-                    </div>
-                {% endif %}
+                            <ul class="abstract-question-list">
+                            {% for rating in review.ratings|sort(attribute='question.position') %}
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: needs moar indentation

Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: `if not rating.question.is_deleted` or `|rejectattr('question.is_deleted')`

File: indico/htdocs/sass/partials/_timelines.scss [hunk 9]
```diff
@@ -63,25 +72,36 @@
         position: relative;
 
         .i-box-header {
+            @extend %time-information;
             // To match i-timeline-item-label height
             min-height: $min-height;
+            justify-content: flex-end;
```
Comment (OmeGak) [inline indico/htdocs/sass/partials/_timelines.scss]: I don't think we want this. `justify-content: flex-end;` on a flex with row direction will align children to the right.

File: indico/htdocs/js/indico/modules/abstracts.js [hunk 10]
```diff
@@ -74,6 +74,9 @@
             $reviewBox.find('.js-edit-review').trigger('click');
             $('body, html').animate({scrollTop: $reviewBox.offset().top}, 'fast');
             $reviewBox.find('.i-timeline-item-box').effect('highlight', {color: Palette.highlight}, 'slow');
+        }).on('click', '.ratings-switch .i-button', function() {
+            $(this).toggleClass('open');
```
Comment (pferreir) [inline indico/htdocs/js/indico/modules/abstracts.js]: Is it really worth it for 2 lines?

File: indico/modules/events/abstracts/templates/abstract/timeline.html (hunk 5 above)
Comment (pferreir) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: Really?

Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: seems strange for me to have it start with an uppercase letter (or does the rule where you can uppercase words in english e.g. in titles also apply for a case like this one)

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss [hunk 11]
```diff
@@ -218,9 +231,24 @@
 }
 
 .abstract-comment-badges {
-    align-self: flex-end;
+    margin-right: auto;
```
Comment (OmeGak) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: I think it's better to use `flex-grow: 1` if you need the div to stretch.

File: indico/htdocs/js/indico/modules/abstracts.js (hunk 10 above)
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/abstracts.js]: in terms of performance it doesn't really make a difference - `$(this)` is quite fast unlike `$('selector')`

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss [hunk 12]
```diff
@@ -218,9 +231,24 @@
...
+    align-self: baseline;
     color: $gray;
-    margin-left: 1em;
+}
+
+.i-timeline-item-metadata {
+    .abstract-metadata-context {
+        align-self: baseline;
+        margin-left: auto;
```
Comment (OmeGak) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: In this case you probably to apply `flex-grow: 1` to `.abstract-metadata-action` instead.

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss [hunk 13]
```diff
@@ -218,9 +231,24 @@
...
 
 .abstract-comment-badges {
-    align-self: flex-end;
+    margin-right: auto;
+    align-self: baseline;
     color: $gray;
-    margin-left: 1em;
+}
+
+.i-timeline-item-metadata {
```
Comment (OmeGak) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: Add `align-items: baseline` and you can remove `align-self: baseline` from `.abstract-metadata-context` and `.abstract-metadata-action`.

File: indico/modules/events/abstracts/templates/abstract/timeline.html [hunk 14]
```diff
@@ -74,33 +76,79 @@
...
+                        </time>
+                        {% if review.modified_dt %}
+                            {{ render_edited_hint(review.modified_dt) }}
+                        {% endif %}
+                    </span>
+                    <span class="abstract-metadata-context">
+                        {{ render_track(review.track, classes='abstract-track-tag i-tag outline') }}
+                    </span>
                 </div>
-                <div class="abstract-comment-badges flex">
```
Comment (OmeGak) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: This breaks hiding the sentence, score and action icons when editing the review.

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss (hunk 11 above)
Comment (pferreir) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: Why? Do I really want to move the div on the right through expanding this one?

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss (hunk 13 above)
Comment (pferreir) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: True!

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss (hunk 11 above)
Comment (OmeGak) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: I think it's better to reason in terms of which element consumes horizontal space without mixing it with margins. 

Using `flex-grow` allows for adding, for instance, `margin-right: 1em` to make sure that no matter how much the content grows there will always be some space in between.

File: indico/modules/events/abstracts/templates/abstract/timeline.html (hunk 5 above)
Comment (OmeGak) [inline indico/modules/events/abstracts/templates/abstract/timeline.html]: I think it looks nice.

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss (hunk 11 above)
Comment (OmeGak) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: It's clearer here. 

---

Using `margin-*` to distribute elements means we lose the ability to use margin for actually adding margins. Example with long content.

screenshot from 2016-11-22 16 10 12

---

`flex-grow: 1` distributes the elements and allows us to keep using `margin-*` to define a margin.

screenshot from 2016-11-22 16 10 36

Comment (pferreir) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: Yeah, that makes sense 👍

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss (hunk 13 above)
Comment (pferreir) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: Or maybe not, because there's a `div.flex` in between the container and the items.

Comment (pferreir) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: Solved.

**Comment by pferreir** [comment]
Superseded by #2722

**Pull request closed without merging**